File: Binary Tree/145. Binary Tree Postorder Traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> postorderTraversal(TreeNode root) {
        List<Integer> arr=new ArrayList<Integer>();
        solve(arr, root);
        return arr;
    }
    void solve(List<Integer> arr, TreeNode root)
    {
        if(root==null)
            return;
        solve(arr,root.left);
        solve(arr,root.right);
        arr.add(root.val);
    }
}

File: Binary Tree/145. Binary Tree Postorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution
{
public:
    vector<int> postorderTraversal(TreeNode *root)
    {
        stack<TreeNode *> st;
        stack<int> t;
        st.push(root);

        while (!st.empty())
        {
            TreeNode *node = st.top();
            st.pop();
            if (node)
                t.push(node->val);

            if (node && node->left)
                st.push(node->left);

            if (node && node->right)
                st.push(node->right);
        }
        vector<int> res;
        while (!t.empty())
        {
            res.push_back(t.top());
            t.pop();
        }
        return res;
    }
};